### DISCUSSION TITLE
current state of data exchange between rust and frontend

### DISCUSSION THREAD

**Discussion by inspired2** [tauri-apps/tauri#3208] [Q&A] [answered]
Hi! Read throw all discussions looking for detailed explanation of dataflow between rust side and frontend, I came across question asked about year ago: https://github.com/tauri-apps/tauri/discussions/1268
Is there currently the way to share data between rust and js? In Electron we have such opportunity with typed arrays(using threads). I have an app that reads and resizes hundreds of images in backend to show them on frontend and having to serde buffers is way too slow. What are currently the alternatives to event messaging with serialization??
Thank you for your work!!

**Comment by FabianLars** [comment, marked as answer]
For this specific use-case _current_ workarounds that come to mind would be to use either a custom protocol, like our streaming example does or via the file system (if saving the images is acceptable) and using `convertFileSrc` to read them or set them as a file source (this uses a custom protocol too).

Otherwise we're still looking at generally improving the serde+transport situation but this won't make it into v1.
